test/test_log_warning.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 Canonical Ltd.
```

Comment:
Nit: it's 2024 :smile:

More importantly, do we want a new test module for this? It seems like a big add for a small feature, and test_log is pretty small (73 lines including whitespace, docs, and comments). I think there's an argument to be made about potentially splitting up some of the very large test files, even though it breaks the symmetry, but I don't think we would start with this one.

Reply:
Thanks, clearly I've copy-pasted this from another file...

Well, my thinking was that we've got some rather long test files (though `test_log.py` is on the shorter end of the spectrum).

I guess I preferred to draft this idea in isolation, and the content can be merged into `test_log.py`.

In the current form, maybe it's easier to review, as I'm using slightly different kind of mocking. I'm open to opinions on this subject 🙇🏻